Tools/clinic/clinic.py

```diff
@@ -2407,12 +2410,22 @@ class Function:
             (not docstring) or ((not docstring[0].isspace()) and (docstring.rstrip() == docstring))
     """
 
-    def __init__(self, parameters=None, *, name,
-                 module, cls=None, c_basename=None,
-                 full_name=None,
-                 return_converter, return_annotation=inspect.Signature.empty,
-                 docstring=None, kind=CALLABLE, coexist=False,
-                 docstring_only=False):
+    def __init__(
+            self,
+            parameters: ParamDict | None = None,
+            *,
+            name: str,
+            module: Module,
+            cls: Class | None = None,
+            c_basename: str | None = None,
+            full_name: str | None = None,
+            return_converter: ReturnConverterType,
+            return_annotation = inspect.Signature.empty,
```

Comment:
This is tricky. I get into a lot of trouble if I annotate this. There's at least one bug uncovered as the result of trying to annotate this one. Some of the havoc may also be incorrect typing introduced earlier.

Reply:
Sounds like something we can defer for now ;)

Reply:
Yes, definitely :) Just a note to self.